signer/src/transaction_coordinator.rs

```diff
@@ -786,6 +856,71 @@ where
         Ok(())
     }
 
+    #[tracing::instrument(skip_all, fields(withdrawal_id = %request.qualified_id()))]
+    async fn construct_and_sign_withdrawal_accept(
+        &mut self,
+        chain_tip: &model::BitcoinBlockRef,
+        wallet: &SignerWallet,
+        bitcoin_aggregate_key: &PublicKey,
+        request: model::SweptWithdrawalRequest,
+    ) -> Result<(), Error> {
+        let stacks = self.context.get_stacks_client();
+        let deployer = self.context.config().signer.deployer;
+
+        let is_completed = stacks
+            .is_withdrawal_completed(&deployer, request.request_id)
+            .await?;
+
+        if is_completed {
+            tracing::warn!("swept withdrawal request already processed");
+            return Ok(());
+        }
+
+        tracing::debug!("processing withdrawal request");
+        let sign_request_fut = self.construct_withdrawal_accept_stacks_sign_request(
+            request,
+            bitcoin_aggregate_key,
+            wallet,
+        );
+
+        let (sign_request, multi_tx) = sign_request_fut.await?;
+        tracing::debug!("constructed withdrawal accept sign request");
+
+        // If we fail to sign the transaction for some reason, we
+        // decrement the nonce by one, and try the next transaction.
+        // This is not a fatal error, since we could fail to sign the
+        // transaction because someone else is now the coordinator, and
+        // all the signers are now ignoring us.
+        let process_request_fut =
+            self.process_sign_request(sign_request, &chain_tip.block_hash, multi_tx, wallet);
+
+        tracing::debug!("processed withdrawal request");
+
+        let status = match process_request_fut.await {
+            Ok(txid) => {
+                tracing::info!(%txid, "successfully submitted accept-withdrawal transaction");
+                "success"
+            }
+            Err(error) => {
+                tracing::warn!(%error, "could not process the stacks sign request for a withdrawal");
+                wallet.set_nonce(wallet.get_nonce().saturating_sub(1));
+                "failure"
+            }
+        };
+
+        metrics::counter!(
+            Metrics::TransactionsSubmittedTotal,
```

Comment:
Not related to this PR just noting, but it wouldn't be bad to add some more specific metrics wrt withdrawal actions

Reply:
I changed this in https://github.com/stacks-network/sbtc/pull/1403/commits/6c31d89b19ec426dc18d9e5d0778e29827fb4b19. I thought about making them constants and but figured that we can do a better job of this in a separate PR.